ReactWindows/ReactNative/UIManager/UIViewOperationQueueCollection.cs

```diff
@@ -0,0 +1,606 @@
+        public void EnqueueCreateView(
+            ThemedReactContext themedContext,
+            int viewReactTag,
+            string viewClassName,
+            ReactStylesDiffMap initialProps,
+            int rootViewTag)
+        {
+            // Called on layout manager thread
+
+            UIViewOperationQueueInstance queue;
+            if (!_reactTagToOperationQueue.TryGetValue(rootViewTag, out queue))
+            {
+                throw new InvalidOperationException("No queue for tag " + rootViewTag);
+            }
+
+            _reactTagToOperationQueue[viewReactTag] = queue;
+
+            queue.EnqueueCreateView(themedContext, viewReactTag, viewClassName, initialProps);
+        }
+
+        /// <summary>
+        /// Enqueue a configure layout animation operation.
+        /// </summary>
+        /// <param name="config">The configuration.</param>
+        /// <param name="success">The success callback.</param>
+        /// <param name="error">The error callback.</param>
+        public void EnqueueConfigureLayoutAnimation(JObject config, ICallback success, ICallback error)
+        {
+            // Called on layout manager thread
+
+            // We have to dispatch this to all queues. Each queue will reset the "animating layout" at the end of the current batch.
+            foreach (var queue in _dispatcherToOperationQueueInfo.Values)
+            {
+                queue.queueInstance.EnqueueConfigureLayoutAnimation(config, success, error);
+            }
+        }
+
+        /// <summary>
+        /// Enqueues an operation to update the properties of a view.
+        /// </summary>
+        /// <param name="tag">The view tag.</param>
+        /// <param name="className">The class name.</param>
+        /// <param name="props">The properties.</param>
+        public void EnqueueUpdateProperties(int tag, string className, ReactStylesDiffMap props)
+        {
+            // Called on layout manager thread
+
+            UIViewOperationQueueInstance queue;
+            if (!_reactTagToOperationQueue.TryGetValue(tag, out queue))
+            {
+                throw new InvalidOperationException("No queue for tag " + tag);
+            }
+
+            queue.EnqueueUpdateProperties(tag, className, props);
+        }
+
+        /// <summary>
+        /// Enqueues an operation to update the layout of a view.
+        /// </summary>
+        /// <param name="parentTag">The parent tag.</param>
+        /// <param name="tag">The view tag.</param>
+        /// <param name="dimensions">The dimensions.</param>
+        public void EnqueueUpdateLayout(
+            int parentTag,
+            int tag,
+            Dimensions dimensions)
+        {
+            // Called on layout manager thread
+
+            UIViewOperationQueueInstance queue;
+            if (!_reactTagToOperationQueue.TryGetValue(tag, out queue))
+            {
+                throw new InvalidOperationException("No queue for tag " + tag);
+            }
```

Comment:
We could put this in a helper so we don't have so many copies of the throw statement.